File: app/src/main/java/com/yhjoo/dochef/utils/ChefAuth.java
package com.yhjoo.dochef.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.google.firebase.auth.FirebaseAuth;
import com.yhjoo.dochef.R;

public class ChefAuth {
    private static FirebaseAuth mAuth;

    public static void LogOut(Context context) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context.getApplicationContext());
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean(context.getString(R.string.SP_ACTIVATEDDEVICE), false);
        editor.remove(context.getString(R.string.SP_USERINFO));
        editor.apply();

        mAuth = FirebaseAuth.getInstance();
        mAuth.signOut();
    }

    public static boolean isLogIn(Context context) {
        mAuth = FirebaseAuth.getInstance();

        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context.getApplicationContext());
        if (sharedPreferences.getBoolean(context.getString(R.string.SP_ACTIVATEDDEVICE), false)
                && mAuth.getCurrentUser() != null) {
            return true;
        } else {
            if (sharedPreferences.getBoolean(context.getString(R.string.SP_ACTIVATEDDEVICE), false)) {
                SharedPreferences.Editor editor = sharedPreferences.edit();
                editor.putBoolean(context.getString(R.string.SP_ACTIVATEDDEVICE), false);
                editor.apply();
            } else if (mAuth.getCurrentUser() != null) {
                mAuth.signOut();
            }
            return false;
        }
    }
}

File: app/src/main/java/com/yhjoo/dochef/utils/ChefAuth.kt
package com.yhjoo.dochef.utils

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatEditText
import androidx.core.content.edit
import androidx.preference.PreferenceManager
import com.google.firebase.auth.FirebaseAuth
import com.yhjoo.dochef.R

object ChefAuth {
    private lateinit var mAuth: FirebaseAuth

    fun logOut(context: Context) {
        Utils.getSharedPreferences(context).edit {
            putBoolean(context.getString(R.string.SP_ACTIVATEDDEVICE), false)
            remove(context.getString(R.string.SP_USERINFO))
            apply()
        }

        mAuth = FirebaseAuth.getInstance()
        mAuth.signOut()
    }

    fun isLogIn(context: Context): Boolean {
        mAuth = FirebaseAuth.getInstance()
        val sharedPreferences = Utils.getSharedPreferences(context)

        return if (sharedPreferences.getBoolean(
                context.getString(R.string.SP_ACTIVATEDDEVICE),
                false
            )
            && mAuth.currentUser != null
        ) {
            true
        } else {
            if (sharedPreferences.getBoolean(
                    context.getString(R.string.SP_ACTIVATEDDEVICE),
                    false
                )
            ) {
                sharedPreferences.edit {
                    putBoolean(context.getString(R.string.SP_ACTIVATEDDEVICE), false)
                    apply()
                }
            } else if (mAuth.currentUser != null) {
                mAuth.signOut()
            }

            false
        }
    }
}